### PR TITLE
Check if malloc / calloc returned NULL

### PR DESCRIPTION
**Pull request by AnHeuermann**
### Related Issues

#917

### Purpose

See if we still get random errors in the testsuite.

### CODE REVIEW

**Comment by AnHeuermann** [comment]
It's quiet hard to test something that is only failing every now and then. Did this improve anything or is it still the same?

File: src/OMSimulatorLib/SystemSC.cpp [hunk 1]
```diff
@@ -204,6 +204,7 @@ oms_status_enu_t oms::SystemSC::instantiate()
       n_states += nStates.back();
       nEventIndicators.push_back(fmus.back()->getNumberOfEventIndicators());
 
+      /* TODO: Will this memory ever get freed??? */
```
Comment (lochel) [inline src/OMSimulatorLib/SystemSC.cpp:207]: Yes, it is done here:

https://github.com/OpenModelica/OMSimulator/blob/94de8fcd4b5dfd53bae9e6f7e4c2925bb017a347/src/OMSimulatorLib/SystemSC.cpp#L396-L403

File: src/OMSimulatorLib/MatVer4.cpp [hunk 2]
```diff
@@ -119,6 +122,11 @@ oms::MatVer4Matrix* oms::readMatVer4Matrix(FILE* file)
   MatVer4Type_t type = (MatVer4Type_t) (matrix->header.type % 100);
   size_t size = sizeofMatVer4Type(type);
   matrix->data = malloc(matrix->header.mrows * matrix->header.ncols * size);
+  if (!matrix->data)
+  {
+    printf("Out of memory");
+    return NULL;
+  }
```
Comment (lochel) [inline src/OMSimulatorLib/MatVer4.cpp:129]: This shouldn't print any error, but the function calling it should check for the return value and issue a warning. But I guess this is just for testing right now.

Comment (AnHeuermann) [reply src/OMSimulatorLib/MatVer4.cpp:129]: But `matrix->data` is used in the next line to read data into it. If we are out of memory this should be a problem.

Comment (lochel) [reply src/OMSimulatorLib/MatVer4.cpp:129]: Yes, you should return, but not print from inside the function.

Comment (AnHeuermann) [reply src/OMSimulatorLib/MatVer4.cpp:129]: Ah, I see.

**Review by lochel — APPROVED** [review]
LGTM :+1:

**Comment by lochel** [comment]
Why does the jenkins report show the asan test being skipped?

**Comment by AnHeuermann** [comment]
I think it is a bug in Blueocean. If I go to the classic pipeline view in Jenkins I can see that the build step of `linux-asan` is evaluated and successful.

**Comment by lochel** [comment]
Ok, then just go ahead and merge this pull request.

**Comment by AnHeuermann** [comment]
We will see if this solves anything. At least the asan test should not be able to run out of memory now. It's using only one thread and has all unlimited virtual memory . If our tests keep failing we can see on which node they are running and at least rule out a memory problem at this place.

**Pull request merged**